### PR TITLE
Select Association View Model

### PR DESCRIPTION
**Pull request by ortoLover**
We have the screen where we select an association. We now need to add it's functionalities, and for that we have to build it's view model and add it to the screen. The functionalities we want for the view model are:

- [x] Clear search field
- [x] Add a search query
- [x] Get the user's name
- [x] Get the list of filtered organizations

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 plugins {
-    id("com.android.application") version "8.2.2" apply false
+    id("com.android.application") version "8.3.2" apply false
```
Comment (Polymeth) [inline build.gradle.kts:3]: Should not be part of this PR. Also, versions above 8.2.2 aren't supported

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #Mon Mar 18 14:31:27 CET 2024
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.2-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.4-bin.zip
```
Comment (Polymeth) [inline gradle/wrapper/gradle-wrapper.properties:4]: Same here about Gradle:)

File: app/src/main/java/com/github/se/assocify/ui/screens/selectAssoc/SelectAssociationViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  private val _uiState: MutableStateFlow<SelectAssociationState> =
+      MutableStateFlow(SelectAssociationState())
+  val uiState: StateFlow<SelectAssociationState>
+
+  init {
+    uiState = _uiState
+    update()
+  }
+
+  private fun update() {
```
Comment (Polymeth) [inline app/src/main/java/com/github/se/assocify/ui/screens/selectAssoc/SelectAssociationViewModel.kt:32]: nit: would change the name because it's unclear what is updated

File: app/src/androidTest/java/com/github/se/assocify/screens/SelectAssociationScreenTest.kt [hunk 4]
```diff
@@ -78,35 +130,89 @@ class SelectAssociationTest : TestCase(kaspressoBuilder = Kaspresso.Builder.with
...
+    }
     ComposeScreen.onComposeScreen<DisplayOrganizationScreenTest>(composeTestRule) {
       organizationName { assertIsDisplayed() }
       organizationIcon { assertIsDisplayed() }
     }
   }
+
+  /* This test check if, when searching with the search bar the icons change */
+  @Test
+  fun testSearchBarWorksWithNoResult() {
```
Comment (Polymeth) [inline app/src/androidTest/java/com/github/se/assocify/screens/SelectAssociationScreenTest.kt:180]: Note for a later PR : would require more testing about the searchbar. Not for now, search bar needs to be uniformized anyway.

File: app/src/androidTest/java/com/github/se/assocify/AssociationAPITest.kt [hunk 5]
```diff
@@ -0,0 +1 @@
+
```
Comment (Polymeth) [inline app/src/androidTest/java/com/github/se/assocify/AssociationAPITest.kt:1]: extremely nit ahaha

File: app/src/main/java/com/github/se/assocify/ui/screens/selectAssoc/SelectAssociationScreen.kt [hunk 6]
```diff
@@ -35,51 +47,77 @@ import androidx.compose.ui.unit.dp
...
+                  modifier = Modifier.testTag("HelloText"),
+                  text = "Hello " + state.value.user.getName() + " !!",
+                  style = MaterialTheme.typography.headlineSmall)
               SearchBar(
                   modifier = Modifier.testTag("SearchOrganization"),
-                  query = "",
-                  onQueryChange = {},
-                  onSearch = {},
+                  query = query,
+                  onQueryChange = { it -> query = it },
```
Comment (Polymeth) [inline app/src/main/java/com/github/se/assocify/ui/screens/selectAssoc/SelectAssociationScreen.kt:73]: nit: no need for `it ->` (implicit lambda)

Comment (ortoLover) [reply app/src/main/java/com/github/se/assocify/ui/screens/selectAssoc/SelectAssociationScreen.kt:73]: Thank you!!

File: app/src/androidTest/java/com/github/se/assocify/AssociationAPITest.kt (hunk 5 above)
Comment (ortoLover) [reply app/src/androidTest/java/com/github/se/assocify/AssociationAPITest.kt:1]: Can fix it

File: app/src/androidTest/java/com/github/se/assocify/screens/SelectAssociationScreenTest.kt (hunk 4 above)
Comment (ortoLover) [reply app/src/androidTest/java/com/github/se/assocify/screens/SelectAssociationScreenTest.kt:180]: Ok thank you for the advice!

**Comment by Polymeth** [comment]
Looks good now! Thank for the work:)

**Review by Polymeth — APPROVED** [review]
